### ISSUE TITLE
Regex context key operator: regex flags ignored

### ISSUE THREAD

**Issue by aeschli** [Microsoft/vscode#42605]
Testing #42331.

I tried the following syntax in our tree-view-sample example
`"when": "view == nodeDependencies && viewItem =~ /Dependency/i"`

It looks like the `i` parameter is ignored.
Just to note. Please close if as designed.

**Comment by jrieken** [comment]
> Just to note. Please close if as designed.

Yeah, not sure. The `i` flag might make sense but the `mg` flags don't really make sense

**Comment by jrieken** [comment]
@alexandrudima Ideas?

**Comment by alexandrudima** [comment]
`i` is probably the only one that makes sense... I don't think we have `\n` in the values to be matched, but `m` would make sense in that case, where it changes the semantics of `\s` or something like that

**Comment by chrisdias** [comment]
I would like this for the docker extension so as to match `Dockerfile`, `dockerfile`, `dockerFile`, etc. Adding Feb in hopes of getting it fixed next month :)
